lib/resources/aws/aws_security_group.rb

```diff
@@ -44,6 +44,7 @@ def allow_only(rules, criteria)
     # allow_{in_out}_only require either a single-rule group, or you
     # to select a rule using position.
     return false unless rules.count == 1 || criteria.key?(:position)
+    return false if criteria.key?(:security_group) && !rules[0][:user_id_group_pairs].count == 1
```

Comment:
I don't think we want to add this case here. If `criteria.key?(:position)` is true, and refers to the second rule, but the first rule does not have a security group reference, this will return a false negative.

Reply:
:-)  Missed that one.
I've added a test for that specific case and updated the code.